src/app/core/models/label.model.ts

```diff
@@ -4,26 +4,27 @@
 export class Label {
```

Comment:
I meant by 
```suggestion
export class Label implements SimplifiedLabel {
```

This way Label is also a simplified label. So in service there is no need to convert the label array into simplified label array as label array is already a simplified label array.

Although we do need to rename the references to SimplifiedLabel from name to formattedName.

Reply:
Let's rename SimplifiedLabel -> SimpleLabel

So it makes sense for Label to implement SimpleLabel as a more complex label

Reply:
I see, that's a much better way of implementing SimplifiedLabel. I've made the changes as suggested in the most recent commit, and also renamed SimplifiedLabel -> SimpleLabel as mentioned.

I've also made some changes to accommodate #183 , as this PR modifies parts of the original `label-filter-bar.component.ts` used in that PR.